### ISSUE TITLE
New Paper: Polar Sparsity

### ISSUE THREAD

**Issue by susavlsh10** [weigao266/Awesome-Efficient-Arch#5]
Hi, thank you for maintaining this excellent survey! 🙌

We would like to suggest adding our recent paper **Polar Sparsity**: High Throughput Batched LLM Inferencing with Scalable Contextual Sparsity to the list.

📄 Paper: https://arxiv.org/abs/2505.14884

💻 Code: https://github.com/susavlsh10/Polar-Sparsity

Polar Sparsity introduces hardware-efficient, sparsity-aware GPU kernels for selective MLP and Attention (attention head sparsity) computations, delivering up to 2.2x end-to-end speedups for models like OPT, LLaMA-2 & 3, Mistral, Qwen across various batch sizes and sequence lengths without compromising accuracy.

Thanks for considering adding this work!

**Comment by weigao266** [comment]
Awesome work! And the concept of Contextual Sparsity is very insteresting. This method seems belonging to the "Mixture of Attention" and "Sparse MoE" category, right?

**Comment by susavlsh10** [comment]
Thank you for your prompt response. The work fits in the following categories:  Dynamic Sparse Attention, Hardware-efficient Implementation, Mixture of Attention, and Sparse MoE. Keep up the awesome work ☺️

**Comment by susavlsh10** [comment]
The review paper appears to be missing recent advances in contextual sparsity.

Contextual sparsity refers to the observation that pre-trained models are inherently sparse, where activating only the most important neurons and attention heads is sufficient to maintain model quality while significantly accelerating inference. It represents the current state-of-the-art in structured sparsity acceleration and is an active and rapidly growing area of research.

Including this line of work would strengthen the paper/repo considerably. I’d be glad to contribute by drafting a section on this topic if that would be helpful. Please let me know your thoughts.

**Comment by weigao266** [comment]
Thank you very much for your active support. It would be great if you could draft a section on "contextual sparsity." Given the structure of the paper, maybe placing it under "Efficient Full Attention" would be appropriate. One concern is: are there many papers on this topic (and it may has some overlap with exist sections)?

BTW, I have already added the Polar Sparsity work to this repository and plan to include it in the next version of the survey paper. Thanks again for your contribution!

**Comment by susavlsh10** [comment]
Hi Weigao, 

I'd be happy to write a draft section on contextual sparsity. I believe there are about 30-40 impactful papers and counting on contextual and activation sparsity. Most works on this area focus on MLP sparsity where each neuron (rows or columns of the weight matrix) are considered an expert. These work are more suited to be placed under "Sparse Mixture-of-Experts" section. Polar Sparsity fits well with Mixture of Attention without requiring full model fine tuning. Let's talk via email :)
